File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int get_gcd(int a, int b){
    if(b == 0)
        return a;
    return get_gcd(b, a % b);
}

int get_lcm(int a, int b){
    return a * b / get_gcd(a, b);
}

vector<int> solution(int n, int m) {
    vector<int> answer;
    int gcd_val = get_gcd(n, m);
    int lcm_val = get_lcm(n, m);
    
    answer.push_back(gcd_val);
    answer.push_back(lcm_val);
    return answer;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    
    public int get_gcd(int a, int b){
        if(b == 0)
            return a;
        return get_gcd(b, a % b);
    }
    
    public int get_lcm(int a, int b){
        return a * b / get_gcd(a, b);
    }
    
    public int[] solution(int n, int m) {
        return new int[]{get_gcd(n, m), get_lcm(n, m)};
    }
}